sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,166 @@
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+
+    def fit(self, *args, **kwargs):
+        self.base_estimator = self.base_estimator.fit(*args, **kwargs)
+        return self
+
+    def _method_name(self):
+        if self.method == 'auto':
+            if getattr(self.base_estimator, 'predict_proba', None):
+                method = 'predict_proba'
+            else:
+                method = 'predict'
+        else:
+            method = self.method
+
+        return method
+
+    def transform(self, *args, **kwargs):
+        t = getattr(self.base_estimator, self._method_name())
+        preds = t(*args, **kwargs)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+    def fit_transform(self, X, y, **fit_params):
+        preds = cross_val_predict(self.base_estimator, X, y, cv=self.cv,
+                                  method=self._method_name(),
+                                  n_jobs=self.n_jobs, fit_params=fit_params)
+
+        self.base_estimator.fit(X, y, **fit_params)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+
+def _identity_transformer():
+    """Contructs a transformer that does nothing"""
+    return FunctionTransformer(lambda x: x)
+
+def make_stack_layer(*base_estimators, **kwargs):
+    """Construct a single layer for a stacked model.
+
+    This is a wrapper around pipelines to provide a more convenient API for
+    stacking models.
+
+    Parameters
+    ----------
+    *base_estimators: list of base estimators.
+
+    restacking: optional, bool, default: False
+        When true, the transformer will return the input.
+
+    **kwargs: Keyword arguments to be passed to `BlendedEstimator`.
+
+    Returns
+    -------
+    f : FeatureUnion with every base estimator wrapped in a
+         `BlendedEstimator`.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.ensemble import make_stack_layer
+    >>> from sklearn.neighbors import KNeighborsClassifier
+    >>> from sklearn.svm import SVC
+    >>> t = make_stack_layer(KNeighborsClassifier(n_neighbors=3), SVC())
+    >>> X = np.array([[1, 3], [.12, 1], [.5, -2], [1, -1], [-2, .1], [7, -84]])
+    >>> y = np.array([1, 0, 0, 1, 0, 1])
+    >>> t.fit_transform(X, y) # doctest: +NORMALIZE_WHITESPACE
+    array([[ 0.66666667,  0.33333333,  0.        ],
+           [ 0.66666667,  0.33333333,  1.        ],
+           [ 0.66666667,  0.33333333,  1.        ],
+           [ 0.66666667,  0.33333333,  0.        ],
+           [ 0.66666667,  0.33333333,  0.        ],
+           [ 0.66666667,  0.33333333,  0.        ]])
+    """
+    restacking = kwargs.pop('restacking', False)
+    estimators = [BlendedEstimator(estimator, **kwargs)
+                  for estimator in base_estimators]
+    if restacking:
+        estimators.append(_identity_transformer())
+    return make_union(*estimators)
```

Comment:
I agree the names are horrrible and should be fixed. about the second part, `FeatureUnion` already exists if someone wants to specify the names. I don't see a reason to create a new api for that. the only arguable point here is that, if someone want's to create a stack layer by hand using `FeatureUnion`, the restacking will have to be implemented by hand as well. I'll give this a thought, but I can't see a clean solution to this yet (other than creating a second function that accepts named estimators instead of unnamed ones).